### PR TITLE
doc: edit README.md - contributing code

### PR DESCRIPTION
**Pull request by zdover23**
Edit the "Contributing Code" section of the "ceph/README.md" file.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
-the license for the modified or contributed file.  Please see the file
-SubmittingPatches.rst for details on what that means and on how to
+The Ceph documentation is licensed under Creative Commons Attribution Share
+Alike 3.0 (CC-BY-SA-3.0). 
+
+Some headers included in the `ceph/ceph` repository are licensed under the GPL.
+See the file `COPYING` for a full inventory of licenses by file.
+
+All code contributions must include a valid "Signed-off-by" line that
+acknowledges the license for the modified or contributed file. See the
```
Comment (anthonyeleven) [inline README.md:19]: I don't think I've ever seen a signed-off-by that mentions a license.

**Review by anthonyeleven — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (zdover23) [reply README.md:19]: @anthonyeleven This is precisely what I had hoped you or someone else would say, ratifying my gut feeling and giving me the license I need to cut this out, secure in my judgment through the unbiased, undirected confirmation of another.

Comment (anthonyeleven) [reply README.md]: Now, if we *should* be mentioning a license explicitly, then we need to add that across the board.  I've never noticed a Ceph Signed-off-by with an explicit license. So I would be inclined to remove this, or re-word it to be *implicitly* approving whatever license model Ceph uses.  The latter is what I was originally told that signed-off-by signifies here.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53049 - Reef backport
https://github.com/ceph/ceph/pull/53050 - Quincy backport

**Pull request merged**